mockboto3/mockboto3/iam/endpoints.py

```diff
@@ -0,0 +1,434 @@
+
+    def list_users(self, kwarg):
+        """List all users."""
+        return responses.list_users_response(self.users)
+
+    def remove_user_from_group(self, kwarg):
+        """Remove user from group if user exists."""
+        self._check_user_exists(kwarg['UserName'], 'RemoveUserFromGroup')
+        self._check_group_exists(kwarg['GroupName'], 'RemoveUserFromGroup')
+
+        group = self.groups[kwarg['GroupName']]
+        user = self.users[kwarg['UserName']]
+
+        group.remove_user(kwarg['UserName'])
+        user.remove_group(kwarg['GroupName'])
+        return responses.generic_response()
+
+    def update_access_key(self, kwarg):
+        try:
+            access_key = self.access_keys.get(kwarg['AccessKeyId'])
+        except KeyError:
+            self._access_key_not_found(kwarg['AccessKeyId'],
+                                       'UpdateAccessKey')
+
+        access_key.status = kwarg['Status']
+        return responses.generic_response()
+
+    def update_login_profile(self, kwarg):
+        """Update login profile for user."""
+        self._check_user_exists(kwarg['UserName'], 'UpdateLoginProfile')
+
+        user = self.users[kwarg['UserName']]
+        self._check_login_profile_exists(user, 'UpdateLoginProfile')
+
+        reset_required = kwarg.get('PasswordResetRequired', None)
+        password = kwarg.get('Password', None)
+        user.update_login_profile(password=password,
+                                  reset_required=reset_required)
+        return responses.generic_response()
+
+    def update_signing_certificate(self, kwarg):
+        """Update signing certificate status."""
+        self._check_user_exists(kwarg['UserName'], 'UpdateSigningCertificate')
+        self._check_signing_certificate_exists(kwarg['UserName'],
+                                               kwarg['CertificateId'],
+                                               'UpdateSigningCertificate')
+
+        user = self.users[kwarg['UserName']]
+        user.update_signing_certificate(kwarg['CertificateId'],
+                                        kwarg['Status'])
+        return responses.generic_response()
+
+    def upload_signing_certificate(self, kwarg):
+        self._check_user_exists(kwarg['UserName'], 'UploadSigningCertificate')
+
+        user = self.users[kwarg['UserName']]
+        for key, cert in user.signing_certs.items():
+            if kwarg['CertificateBody'] == cert.body:
+                raise client_error('UploadSigningCertificate',
+                                   'DuplicateCertificate',
+                                   'A duplicate certificate already exists.')
+
+        cert = user.upload_signing_certificate(kwarg['CertificateBody'])
+        return responses.upload_signing_certificate_response(kwarg['UserName'],
+                                                             cert)
+
+    def _check_user_exists(self, user, method):
```

Comment:
In the absence of any logical closeness needs please sort methods in alpha order